tests/unit/preprocessor/_mask/test_mask_multimodel.py

```diff
@@ -0,0 +1,315 @@
+"""Unit tests for :func:`esmvalcore.preprocessor.mask_multimodel`."""
+
+from unittest import mock
+
+import iris
+import numpy as np
+import pytest
+
+from esmvalcore.preprocessor import PreprocessorFile as PreprocessorFileBase
+from esmvalcore.preprocessor._mask import (
+    _get_shape,
+    _multimodel_mask_cubes,
+    _multimodel_mask_products,
+    mask_multimodel,
+)
+
+
+class PreprocessorFile(mock.Mock):
+    """Mocked PreprocessorFile."""
+
+    def __init__(self, cubes, filename, **kwargs):
+        """Initialize with cubes."""
+        super().__init__(spec=PreprocessorFileBase, **kwargs)
+        self.filename = filename
+        self.cubes = cubes
+        self.mock_ancestors = set()
+        self.wasderivedfrom = mock.Mock(side_effect=self.mock_ancestors.add)
+
+
+def assert_array_equal(array_1, array_2):
+    """Assert that (masked) array 1 equals (masked) array 2."""
+    if np.ma.is_masked(array_1) or np.ma.is_masked(array_2):
+        np.testing.assert_array_equal(np.ma.getmaskarray(array_1),
+                                      np.ma.getmaskarray(array_2))
+        mask = np.ma.getmaskarray(array_1)
+        np.testing.assert_array_equal(array_1[~mask], array_2[~mask])
+    else:
+        np.testing.assert_array_equal(array_1, array_2)
```

Comment:
just use the `assert_array_equal` func from our test library, man, no need to reinvent the wheel

Reply:
1. This function can't be used for scalars, e.g., `AttributeError: 'int' object has no attribute 'mask'`.
2. It fails for arrays like `a = np.ma.masked_equal([0, 1], 0)` and `b = np.ma.masked_equal([1, 1], 1)` which it shouldn't do in my opinion. Both arrays are essentially `[-- 1]`.

I don't like to change the function in the library, maybe some other tests depend on this specific behavior and I don't wanna mess around with that.

Reply:
hmm maybe we should develop that function a bit more then, anyways not in this PR so fine by me :+1: